Swift/Tests/PublisherTest.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  PublisherTest.swift
+//  CouchbaseLite
+//
+//  Copyright (c) 2025 Couchbase, Inc. All rights reserved.
+//
+//  Licensed under the Couchbase License Agreement (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//  https://info.couchbase.com/rs/302-GJY-034/images/2017-10-30_License_Agreement.pdf
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Combine
+import XCTest
+@testable import CouchbaseLiteSwift
+
+@available(iOS 13.0, *)
+class PublisherTest: CBLTestCase {
```

Comment:
I think it might be easier to put separate tests into the CollectionTest, ReplicatorTest, and QueryTest class especially the publishers for the Replicator, otherwise this test class will need to bring several things from the ReplicatorTest. Also the tests can stay closer to the current add/remove change listener tests as they are in the same category.